Library/Homebrew/dev-cmd/pr-pull.rb

```diff
@@ -209,22 +209,21 @@ def squash_formula_commits(commits, file, reason: "", verbose: false, resolve: f
     ohai bump_subject
   end
 
-  def autosquash!(original_commit, path: ".", reason: "", verbose: false, resolve: false)
-    path = Pathname(path).extend(GitRepositoryExtension)
-    original_head = path.git_head
+  def autosquash!(original_commit, tap:, reason: "", verbose: false, resolve: false)
+    original_head = tap.path.git_head
 
-    commits = Utils.safe_popen_read("git", "-C", path, "rev-list",
+    commits = Utils.safe_popen_read("git", "-C", tap.path, "rev-list",
                                     "--reverse", "#{original_commit}..HEAD").lines.map(&:strip)
 
     # Generate a bidirectional mapping of commits <=> formula files.
     files_to_commits = {}
     commits_to_files = commits.to_h do |commit|
-      files = Utils.safe_popen_read("git", "-C", path, "diff-tree", "--diff-filter=AMD",
+      files = Utils.safe_popen_read("git", "-C", tap.path, "diff-tree", "--diff-filter=AMD",
                                     "-r", "--name-only", "#{commit}^", commit).lines.map(&:strip)
       files.each do |file|
         files_to_commits[file] ||= []
         files_to_commits[file] << commit
-        next if %r{^Formula/.*\.rb$}.match?(file)
+        next if (tap.path/file).dirname == tap.formula_dir && File.extname(file) == ".rb"
```

Comment:
Does this not work?

```suggestion
        next if tap.path == tap.formula_dir && File.extname(file) == ".rb"
```

Reply:
But doesn't `(tap.path/file).dirname` just return `tap.path`?

Reply:
Not if `file` is `Formula/a.rb`, in which case it would be `tap.path/"Formula"`.

I could perhaps write this as `tap.path/File.dirname(file)` instead.

Reply:
Gotcha forgot that was a possibility. No need to change imo